.github/workflows/build-gradle.yml

```diff
@@ -0,0 +1,100 @@
+name: Build
+
+on:
+  workflow_call:
+    secrets:
+      codeartifact-username:
+        description: The username to access the codeartifact repository.
+        required: false
+      codeartifact-password:
+        description: The password to access the codeartifact repository.
+        required: false
+      sonar-token:
+        description: A token allowing access to SonarCloud.
+        required: true
+
+jobs :
+  validate-wrapper:
+    name: Validate Gradle wrapper
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Gradle wrapper validation
+        uses: gradle/wrapper-validation-action@v1.0.4
+
+  build:
+    name: Build
+    needs: validate-wrapper
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK
+        uses: actions/setup-java@v2.3.1
+        with:
+          cache: gradle
+          distribution: temurin
+          java-version: 17
+
+      - name: Check for codeartifact secrets
+        id: check-codeartifact-secrets
+        env:
+          USERNAME: ${{ secrets.codeartifact-username }}
+          PASSWORD: ${{ secrets.codeartifact-password }}
+        run: |
+          if [[ $USERNAME != "" && $PASSWORD != "" ]]; then
+            echo ::set-output name=do-codeartifact-login::true
+          else
+            echo ::set-output name=do-codeartifact-login::false
+          fi
+
+      - name: Configure AWS credentials
+        if: steps.check-codeartifact-secrets.outputs.do-codeartifact-login == 'true'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.codeartifact-username }}
+          aws-secret-access-key: ${{ secrets.codeartifact-password }}
+          aws-region: eu-west-1
```

Comment:
This region looks correct

Reply:
@ReubenRobertsHEE I'm not sure if you're confirming or questioning 😂 

Our current codeartifact repo only exists in eu-west-1, another case of it not being supported in eu-west-2 when we needed to use it.

Reply:
Yes, I was confirming - sorry, just working though the scripts and making sure I understand and have checked what I can :)